### PR TITLE
Introduce Upstream fetch timeouts

### PR DESCRIPTION
**Pull request by andrewscfc**
Resolves https://github.com/bbc/simorgh-infrastructure/issues/1235

**Overall change:**
Introduces a timeouts to make Simorgh more resiliant to upstream fetches taking too long to respond making it more likely a page is still returned, even without 'secondary content'. I am defining 'secondary content' as content that supports the main content of the page but is not fundamental to user's experience e.g. Recommendations. I am defining 'primary content' as the main content of the page such as the blocks for a Story page, Media page etc.

- Changes all page data fetches to use timeout of 4500ms matching mozarts timeout:
image
  - This ensures Simorgh does not hang waiting to receive page data after Mozart has disconnected and returned a timeout response
- Change most 'secondary data' fetches to use a timeout of 2000ms
  - This is for all 'additional data' fetches
  - The upstream fetch for includes
  - The fetch for toggles is not included in this PR, I feel this should be included in a follow-up PR to not change too much in one PR https://github.com/bbc/simorgh/blob/latest/src/app/lib/utilities/getToggles/index.js#L35 Please feel free to discuss thi
---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [ ] I have assigned this PR to the Simorgh project

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

**Comment by andrewscfc** [comment]
Following the roll out of this PR, we can design queries identifying if the secondary timeout is too aggressive, this existing log can be used for this purpose: https://github.com/bbc/simorgh/pull/8120/files#diff-2f6af00c0fdb21e32f5a4068954f1a455ae07ea0499aa290a8d8f9af89626f96R87-R93

@chris-hinds do you feel this is adequate for monitoring or should we add custom cloudwatch metrics? I'm not sure how we do this?

File: scripts/bundleSize/bundleSizeConfig.js [hunk 1]
```diff
@@ -5,5 +5,5 @@ module.exports = {
   // below the smallest value in the build output; this avoids the
   // need for frequent changes as bundle sizes fluctuate.
   MIN_SIZE: 673,
-  MAX_SIZE: 851,
+  MAX_SIZE: 855,
```
Comment (andrewscfc) [inline scripts/bundleSize/bundleSizeConfig.js]: the bundlesize has tipped over into 852, I've put it up to 855, in keeping with expectation to be around 5 kb the actual bundle size

**Comment by andrewscfc** [comment]
@chris-hinds @rhenshaw56 @Louis-Matsika - you may notice the primary timeout is 4500ms in this PR rather than 3000ms from our meeting, I double checked the timeout using the mozart debugging url and found the Simorgh timeout is actually 4500ms, hence the change.

**Review by rhenshaw56 — APPROVED** [review]

File: src/app/lib/utilities/getFetchTimeouts/index.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+export const PRIMARY_DATA_TIMEOUT = 4500;
+
+/**
+ * This timeout is for fetching 'secondary' data that is not
+ * so fundamental to delivering a meaningful response to the user.
+ * If a timeout occurs retrieving this data, a page can still be
+ * constructed without the data being retrieved and returned to
+ * be served via Mozart.
+ */
+export const SECONDARY_DATA_TIMEOUT = 2000;
```
Comment (karinathomasbbc) [inline src/app/lib/utilities/getFetchTimeouts/index.js:22]: Probably overkill, do we want to add a snapshot test for these values? e.g. https://github.com/bbc/simorgh/blob/latest/src/app/lib/styles.const.test.js

File: src/app/routes/utils/fetchPageData/index.js [hunk 3]
```diff
@@ -35,10 +36,14 @@ export const getUrl = pathname => {
 /**
  * An isomorphic fetch wrapper for pages, with error and log handling.
  * @param {string} path The URL of a resource to fetch.
+ * @param {number} timeout Optional parameter to provide a custom timeout
+ * for request for 'secondary data'. The fetch timeout defaults to the 'primary
+ * data' timeout if this is not provided.
```
Comment (karinathomasbbc) [inline src/app/routes/utils/fetchPageData/index.js:41]: Is it worth providing a link to the value of the primary data timeout here?

**Review by karinathomasbbc — APPROVED** [review]

**Review by chris-hinds — APPROVED** [review]

**Review by jroebu14 — APPROVED** [review]

File: src/app/lib/utilities/getFetchTimeouts/index.js (hunk 2 above)
Comment (andrewscfc) [reply src/app/lib/utilities/getFetchTimeouts/index.js:22]: I'm not really sure what snapshot tests add in this case?

File: src/app/routes/utils/fetchPageData/index.js (hunk 3 above)
Comment (andrewscfc) [reply src/app/routes/utils/fetchPageData/index.js:41]: I've added the link in this commit: https://github.com/bbc/simorgh/pull/8120/commits/450dd8572b5194ff708a5dc4dd6f8ed2906e7691
The link won't work until we merge to latest however

**Pull request merged**